### PR TITLE
Add Attachment tweaks

### PR DESCRIPTION
**Pull request by mhdostal**
This improves adding attachments from files and the photo library.

- remove workaround for Apollo 635
- ensure Files added on a device use `SecurityScopedResource` methods
- allows all file types and all photo picker types to be added as attachments

### CODE REVIEW

File: Sources/ArcGISToolkit/Common/AttachmentModel.swift [hunk 1]
```diff
@@ -90,25 +90,8 @@ import SwiftUI
...
-                
-                // WORKAROUND - attachment.fileURL is just a GUID for FormAttachments
-                // Note: this can be deleted when Apollo #635 - "FormAttachment.fileURL is not user-friendly" is fixed.
-                var tmpURL = attachment.fileURL
-                tmpURL = tmpURL?.deletingLastPathComponent()
-                tmpURL = tmpURL?.appending(path: attachment.name)
-                
-                _ = FileManager.default.secureCopyItem(at: attachment.fileURL!, to: tmpURL!)
-                url = tmpURL!
-            }
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Common/AttachmentModel.swift:108]: Hmm, I thought I already pulled these out in [715](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/715/files). Maybe this branch was just behind? Not sure why the change is still showing as a diff though.

File: Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentPhotoPicker.swift [hunk 2]
```diff
@@ -32,7 +32,10 @@ struct AttachmentPhotoPicker: ViewModifier {
             .photosPicker(
                 isPresented: $photoPickerIsPresented,
                 selection: $item,
-                matching: .any(of: [.images, .not(.livePhotos)])
+                matching: .any(of: [
+                    .images,
+                    .videos
+                ])
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentPhotoPicker.swift:38]: Do we still want to disallow live photos?

Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentPhotoPicker.swift:38]: Based on your description I assume this was intended.

Comment (mhdostal) [reply Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentPhotoPicker.swift:38]: Yes, intended. I tried in on-device and it worked. I think it simply took the "key frame" and used that.  Most of the image/video types I tried worked successfully, with the exception of videos that come in with the `.quicktime` extension, which don't get applied to the database, but do show up in the in-memory list of attachments.

Comment (mhdostal) [reply Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentPhotoPicker.swift:38]: There will be another PR listing the issues I've encountered.

File: Sources/ArcGISToolkit/Common/AttachmentModel.swift (hunk 1 above)
Comment (mhdostal) [reply Sources/ArcGISToolkit/Common/AttachmentModel.swift]: I merged the `FeatureFormAttachments` branch into this branch and that went away. Sorry I didn't do that first.

Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Common/AttachmentModel.swift]: Ah nice, no worries it just had me confused for a moment.

File: Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentImportData.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// limitations under the License.
+
+import ArcGIS
+import OSLog
+import SwiftUI
+
+/// Data used to create the attachment.
+struct AttachmentImportData: Equatable {
+    var data: Data
+    var contentType: String
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentImportData.swift:22]: Should the first two properties be constants?

```suggestion
    let data: Data
    let contentType: String
```

**Review by dfeinzimer — APPROVED** [review]
Just the one question.

**Review by dfeinzimer — APPROVED** [review]
